test/digitalTwin/utility.test.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { Utility } from "../../src/DigitalTwin/pnp/src/common/utility";
+import { Constants } from "../../src/DigitalTwin/pnp/src/common/constants";
+import { ModelType } from "../../src/DigitalTwin/pnp/src/deviceModel/deviceModelManager";
+
+const { readFile, writeJson, pathExists } = require("fs-extra");
+
+jest.mock("fs-extra");
+
+describe("Utility", () => {
+  const folder = "root";
+  const name = "test";
+  const modelId = "urn:model:id:1";
+  const replacement = new Map<string, string>();
+  replacement.set(Constants.DIGITAL_TWIN_ID_PLACEHOLDER, modelId);
+  replacement.set("{foo}", "bar");
+
+  test("create file from template", async () => {
+    const templatePath = "templatePath";
+    const filePath = "filePath";
+    const writeOptions = {
+      spaces: Constants.JSON_SPACE,
+      encoding: Constants.UTF8
+    };
+    const template = '{"@id": "{DigitalTwinIdentifier}"}';
```

Comment:
>DigitalTwinIdentifier [](start = 32, length = 21)

Can we replace this with Constants.DIGITAL_TWIN_ID_PLACEHOLDER?

Reply:
Here I want to make template string be more readable/straightforward instead of concatenating one string and one variable.

Reply:
fixed